testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/bmt/timeout/StatelessBmtBean.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat Middleware LLC, and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.test.integration.ejb.transaction.bmt.timeout;
+
+import java.rmi.RemoteException;
+import javax.annotation.Resource;
+import javax.ejb.EJBException;
+import javax.ejb.Stateless;
+import javax.ejb.TransactionManagement;
+import javax.ejb.TransactionManagementType;
+import javax.naming.NamingException;
+import javax.transaction.RollbackException;
+import javax.transaction.SystemException;
+import javax.transaction.TransactionManager;
+import javax.transaction.UserTransaction;
+
+import org.jboss.as.test.integration.transactions.TxTestUtil;
+import org.jboss.logging.Logger;
+
+@Stateless
+@TransactionManagement(TransactionManagementType.BEAN)
+public class StatelessBmtBean {
+    private static Logger log = Logger.getLogger(StatelessBmtBean.class);
+
+    @Resource
+    private UserTransaction txn;
+
+    @Resource(name = "java:jboss/TransactionManager")
+    private TransactionManager tm;
+
+    public String testTransaction(int transactionCount, int timeoutCount) throws RemoteException, NamingException {
+        TxTestUtil.checkTransactionExists(tm, false);
+
+        for(int i = 1; i <= transactionCount; i++) {
+            boolean isTimeout = (timeoutCount - i >= 0) ? true : false;
```

Comment:
Why not simply?
    (timeoutCount - i >= 0)
or simpler:
    i <= timeoutCount

And in a simple test ver see this becoming